kudu-handler/src/test/queries/positive/kudu_queries.q

```diff
@@ -0,0 +1,148 @@
+SELECT key FROM all_types_table WHERE bool >= true;
+SELECT key FROM all_types_table WHERE bool < false;
+SELECT key FROM all_types_table WHERE bool <= false;
+
+-- Verify comparison predicates on string.
+-- Note: string is escaped because it's a reserved word.
+SELECT key FROM all_types_table WHERE `string` = "one";
+SELECT key FROM all_types_table WHERE `string` != "one";
+SELECT key FROM all_types_table WHERE `string` > "one";
+SELECT key FROM all_types_table WHERE `string` >= "one";
+SELECT key FROM all_types_table WHERE `string` < "two";
+SELECT key FROM all_types_table WHERE `string` <= "two";
+
+-- Verify comparison predicates on binary.
+-- Note: binary is escaped because it's a reserved word.
+SELECT key FROM all_types_table WHERE `binary` = cast ('one' as binary);
+SELECT key FROM all_types_table WHERE `binary` != cast ('one' as binary);
+SELECT key FROM all_types_table WHERE `binary` > cast ('one' as binary);
+SELECT key FROM all_types_table WHERE `binary` >= cast ('one' as binary);
+SELECT key FROM all_types_table WHERE `binary` < cast ('two' as binary);
+SELECT key FROM all_types_table WHERE `binary` <= cast ('two' as binary);
+
+-- Verify comparison predicates on timestamp.
+-- Note: timestamp is escaped because it's a reserved word.
+SELECT key FROM all_types_table WHERE `timestamp` = '2011-11-11 11:11:11';
+SELECT key FROM all_types_table WHERE `timestamp` != '2011-11-11 11:11:11';
+SELECT key FROM all_types_table WHERE `timestamp` > '2011-11-11 11:11:11';
+SELECT key FROM all_types_table WHERE `timestamp` >= '2011-11-11 11:11:11';
+SELECT key FROM all_types_table WHERE `timestamp` < '2012-12-12 12:12:12';
+SELECT key FROM all_types_table WHERE `timestamp` <= '2012-12-12 12:12:12';
+
+-- Verify comparison predicates on decimal.
+-- Note: decimal is escaped because it's a reserved word.
+SELECT key FROM all_types_table WHERE `decimal` = 1.111;
+SELECT key FROM all_types_table WHERE `decimal` != 1.111;
+SELECT key FROM all_types_table WHERE `decimal` > 1.111;
+SELECT key FROM all_types_table WHERE `decimal` >= 1.111;
+SELECT key FROM all_types_table WHERE `decimal` < 2.222;
+SELECT key FROM all_types_table WHERE `decimal` <= 2.222;
+
+-- Verify null predicates.
+SELECT key FROM all_types_table WHERE null IS NOT NULL;
+SELECT key FROM all_types_table WHERE null IS NULL;
+
+-- Verify AND predicates.
+SELECT key FROM all_types_table WHERE null IS NULL AND key > 1;
+SELECT key FROM all_types_table WHERE null IS NULL AND key >= 1;
+
+-- Verify OR predicates.
+SELECT key FROM all_types_table WHERE key = 1 OR `string` = "one";
+SELECT key FROM all_types_table WHERE key = 1 OR `string` = "two";
+
+-- Verify various other filters.
+SELECT key FROM all_types_table WHERE string IN ("one", "missing");
+SELECT key FROM all_types_table WHERE string NOT IN ("one", "missing");
+SELECT key FROM all_types_table WHERE key BETWEEN -1 and 1;
+SELECT key FROM all_types_table WHERE `string` LIKE "%n%";
+
```

Comment:
Can we add a few more queries to the tests (maybe in a new q file)?
In particular it would be interesting to execute more complex operations on top of Kudu table so all computation is not pushed to it from Hive (and include a few explain plans).
Another helpful example would be creating a second table in Hive and try to join any of the Kudu-backed tables with it (and maybe add some operations on top, e.g., group by).
In the past we have discovered some bugs with custom storage handlers when more complex query processing was involved.

Reply:
I added a couple queries with join, group by, and order by.